src/api/routes/application.js

```diff
@@ -11,15 +14,36 @@ export default (app) => {
     /**
      * Creates a new Company Application
      */
-    router.post("/", validators.create, async (req, res, next) => {
-
+    router.post("/", validators.create, applicationMiddleware.exceededCreationTimeLimit, async (req, res, next) => {
         try {
+            const applicationService = new ApplicationService();
+            await applicationService.deleteApplications(req.body.email);
             // This is safe since the service is destructuring the passed object and the fields have been validated
-            const application = await (new ApplicationService()).create(req.body);
+            const application = await applicationService.create(req.body);
```

Comment:
We don't need to delete the application, just "update or create"
https://stackoverflow.com/a/72043808